src/types/utilities.ts

```diff
@@ -47,7 +47,11 @@ export function isFalsyType(type: ts.Type): boolean {
 	}
 
 	if (type.isLiteral()) {
-		return !type.value;
+		if (typeof type.value === "object") {
+			return type.value.base10Value === "0";
```

Comment:
Would also be completely possible to do convert to an actual bigint, i.e.
`return !BigInt((type.value.negative ?  "-" : "") + type.value.base10Value)`.
However, we don't actually need the bigint value here, so 🤷

Reply:
Yeah 🤷 I'm more motivated to keep this simple than anything else.